python/pyspark/ml/torch/data.py

```diff
@@ -0,0 +1,78 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch
+import numpy as np
+
+
+class SparkPartitionTorchDataset(torch.utils.data.IterableDataset):
+
+    def __init__(self, arrow_file_path, schema, num_samples):
+        self.arrow_file_path = arrow_file_path
+        self.num_samples = num_samples
+        self.field_types = [field.dataType.simpleString() for field in schema]
```

Comment:
why we need this? arrow data itself already contains schema

Reply:
> arrow data itself already contains schema

Some information is lost. e.g. For VectorUDT type, here we get struct type for it.